pkg/azure/provider/provider.go

```diff
@@ -79,10 +80,18 @@ func (d defaultDriver) CreateMachine(ctx context.Context, req *driver.CreateMach
 		return
 	}
 
-	vm, err := helpers.CreateVM(ctx, d.factory, connectConfig, providerSpec, imageReference, plan, req.Secret, nicID, vmName)
+	// create disks with image ref since they can not be created together with the vm
+	// TODO parallelize creation with nic?
+	imageRefDiskIDs, err := helpers.CreateDisksWithImageRef(ctx, d.factory, connectConfig, providerSpec, vmName)
```

Comment:
Since now there are 2 resources that gets created before the VM gets created. Therefore it is possible that while the NIC and disks with Image reference are successfully created, the VM creation fails. Have you tested that when a DeleteMachine gets called, both NICs and Disks with Image References gets deleted? As in this case there will not be any VM.

Also in the `triggerDeletion` flow in MCM, there is an issue https://github.com/gardener/machine-controller-manager/issues/936 where we need to remove call to `getMachineStatus`. Since this adds another set of resources that are now orphan, we would like to fix MCM #936 issue first before this can go in.